pkg/gas/models.go

```diff
@@ -362,13 +362,23 @@ func (e *evmFeeEstimator) estimateFeeLimit(ctx context.Context, feeLimit uint64,
 	if !e.geCfg.EstimateLimit() {
 		return providedGasLimit, nil
 	}
+
+	// If EstimationSenderAddress string is not empty, then we unwrap the config value and set the pointer. Otherwise, it remains nil
+	var estimationSenderAddress *common.Address
+	if len(*e.geCfg.EstimationSenderAddress()) != 0 {
```

Comment:
Think we should check for nil here before dereferencing for safety
```suggestion
	if e.geCfg.EstimationSenderAddress() != nil && len(*e.geCfg.EstimationSenderAddress()) != 0 {
```

Reply:
Thank you Amit! I have just added this check